util/dao/utils/src/lib.rs

```diff
@@ -25,36 +27,55 @@ pub enum Error {
     Format,
 }
 
-pub fn genesis_dao_data(genesis_cellbase_tx: &TransactionView) -> Result<Byte32, FailureError> {
-    let c = genesis_cellbase_tx
-        .data()
-        .raw()
-        .outputs()
-        .into_iter()
-        .try_fold(Capacity::zero(), |capacity, output| {
-            let cap: Capacity = output.capacity().unpack();
-            capacity.safe_add(cap)
-        })?;
-    let u = genesis_cellbase_tx
-        .data()
-        .raw()
-        .outputs()
-        .into_iter()
-        .zip(
-            genesis_cellbase_tx
-                .data()
-                .raw()
-                .outputs_data()
-                .into_iter()
-                .map(|d| d.raw_data()),
-        )
-        .try_fold(Capacity::zero(), |capacity, (output, data)| {
-            Capacity::bytes(data.len()).and_then(|data_capacity| {
-                output
-                    .occupied_capacity(data_capacity)
-                    .and_then(|c| capacity.safe_add(c))
+pub fn genesis_dao_data(txs: Vec<&TransactionView>) -> Result<Byte32, FailureError> {
+    let dead_cells = txs
+        .iter()
+        .flat_map(|tx| tx.inputs().into_iter().map(|input| input.previous_output()))
+        .collect::<HashSet<_>>();
+    let statistics_outputs = |tx: &TransactionView| -> Result<_, FailureError> {
+        let c = tx.data().raw().outputs().into_iter().try_fold(
+            Capacity::zero(),
+            |capacity, output| {
+                let cap: Capacity = output.capacity().unpack();
+                capacity.safe_add(cap)
+            },
+        )?;
```

Comment:
`c` should substrate `output.capacity()` if the output cell was dead, just like `u`.

Reply:
I have written a test for it on my branch. 
Please let me know after fixing this problem, or if I am wrong.

test: https://github.com/keroro520/ckb/blob/7a198fc7729bac9181348da9851d70696890804a/test/src/specs/reward/primary_reward.rs#L108-L159

Reply:
I filtered out dead cells below (by not add it).
``` rust
.filter(|(index, _)| {
    !dead_cells.contains(&OutPoint::new(tx_hash.clone(), *index as u32))
})
```

Reply:
we have to filter out dead cells in both c and u.

Reply:
> we have to filter out dead cells in both c and u.

In common blocks we count every cell to `c`.
So, filter out dead cells in `c` is a genesis specific logic?

Reply:
In common blocks, we should filter out dead cells as well.

`c` is the total issued capacity, is the sum capacity of ***all the live cells***.

Reply:
> > we have to filter out dead cells in both c and u.
> 
> In common blocks we count every cell to `c`.
> So, filter out dead cells in `c` is a genesis specific logic?

C should only count the outputs in the cellbase. And u should count occupied capacity in all live cells.

Reply:
> `c` is the total issued capacity, is the sum capacity of _**all the live cells**_.

Unless no shannons were burned.